### ISSUE TITLE
Analytics Feature Request | Track Google Translate | C&L Team

### ISSUE THREAD

**Issue by SKasimow** [department-of-veterans-affairs/va.gov-team#28943]
# Analytics Feature Request

## What this form is for

Use this template to request an Analytics dashboard feature or report a dashboarding or Google Analytics/Google Tag Manager bug. This form is for enhancements to the existing KPI dashboards. If you have any analytics questions, please use the `/support` command on [#vfs-platform-support](https://dsva.slack.com/archives/CBU0KDSB1) channel in Slack, or attend one of our [office hours](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Analytics-customer-support-guidelines.1586823275.html#Analyticscustomersupportguidelines-5.AdditionalAnalyticsSupport).

Platform Analytics & Insights will use the information you provide to help improve our products and services.

## Instructions

Please fill out the following sections:

- [ ] Ticket Description
- [ ] Request/Report Description
- [ ] Additional Comments

## Ticket Description

Who is this request for?

- What is your role, and which team/project is this work for? (optional for feature requests, required for bug reports)

Request from the Content and Localization team for the Translation Recommendation project.

  @SKasimow 

- Is this a feature request or a bug?

  - [x] Feature request
  - [ ] Bug report

## Request/Report Description

Please describe the feature you are requesting or bug you are reporting. Please be as detailed as possible. 

For a new feature request:

- Description of desired feature

We would like to capture when a visitor uses Google/Chrome ("Translate to") or Bing Translate to translate a VA.gov page to another language while still on the page. By using a Page Listener to track events we should be able to capture these occurrences. 

**Update 9/21:** Additionally, if we can capture and track events when Google Translate (outside VA.gov) is used with minimal effort, that would be a nice-to-have. In this scenario, when the URL is clicked, the translated page will be opened in an iFrame

![Screen Shot 2021-09-21 at 3.58.04 PM.png](https://images.zenhubusercontent.com/5fb57aafa11eae0592a15cb1/1c98e461-794d-437e-8a65-881b6befd90b)

Examples detailed here: https://www.analyticsmania.com/post/track-page-translations-with-google-tag-manager-and-google-analytics/ 

- Why the feature is needed

As part of upcoming Congressional update on Translation Recommendations we need to understand the frequency a LEP visitor needs translation support to read one of our pages.

- (Data) What timeframes are you looking to analyze (e.g. before vs. after dd/mm/yy)?

Ideally we would like to have this in place by end of August if possible in order to capture a full month of usage before presentation to Congress.

**Product questions to think through if you're requesting new data features:**

- Who is your audience?
C+L team, leadership, Congress

- Why do you need this data?
Provide better insight into LEP visits/visitors that need translation support and help size the audience by language and prioritize pages/sections for translation.

- What is your hypothesis that you're hoping the data will support?
There is a larger base of users visiting VA.gov that need language support than we can see just by looking at browser settings for language.

- What area of the VA are you hoping to assess?
Need to discuss with Analytics how to approach this request; is it better to break into phases vs implementing site-wide?

- How frequently do you need this data?
Would like to be able to gather daily/weekly/monthly as needed.

## Additional Comments

- Should this be a very limited release in order to validate performance impact before expanding to top xx pages?
- Need to capture which pages are translated and what languages they are translated to

> Please leave the following blank

## Acceptance Criteria

- [ ] Relabel the issue where appropriate
- [ ] Follow up with the requester where necessary

For features:

- [ ] Groomed & backlogged with notes

## Definition of Done

- [ ] All AC completed

**Comment by bsmartin-ep** [comment]
@jennaouellet -

I _think_ you volunteered to QA this.

GTM workspace:
https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000555

This uses a weird [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to track browser-based Google translations.

1. Do you see any noticeable change in page performance from this tag?
2. Event Action should be like `Page Translate ~ google ~ es`.  Should we try to get the original page language or can that be inferred by the page URL? (Maybe a question for @SKasimow?)

**Comment by jennaouellet** [comment]
@bsmartin-ep 

The load time did increase a bit, here are the ranges I'm getting from the console:

Normal load time - 1.26s - 1.97s
Load time with translate tracking - 2.64s - 3.04s

As an end user there was not a noticeable lag on my end between the two. 

The Lighthouse reports showed negligible differences; some of that is due to the GTM Preview's load on top of the page itself.

**Comment by bsmartin-ep** [comment]
Thanks @jennaouellet.  Is "normal load time" also running through the GTM preview (of the default workspace I guess)?

**Comment by jennaouellet** [comment]
@bsmartin-ep that's correct - those lower times are from the default workspace preview

**Comment by SKasimow** [comment]
@bsmartin-ep If we are able to capture the original language of the page, that would be a nice-to-have.

**Comment by SKasimow** [comment]
Hi @bsmartin-ep - is performance acceptable? Should we limit release to prod and test for a few days before going wide?

And can you tell me how I can pull data once implemented?

Thanks :)

Sharon

**Comment by SKasimow** [comment]
@bsmartin-ep Hi Brian - I added an update to the ticket for tracking Google Translate service (outside VA.gov) if small LOE.

I think the more critical piece is seeing whether we can improve performance or have Tools team better quantify impacts before we can implement, even on a sub-set of pages.

Thanks!

Sharon

**Comment by bsmartin-ep** [comment]
We're on staging @SKasimow.

These events - event label `page_translate` and event action begins with `Page Translate ~` - should start filtering in to the non-prod property:

https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w178298118p176884211/_u.date00=20210927&_u.date01=20210927&explorer-segmentExplorer.segmentId=analytics.eventAction&explorer-table.plotKeys=%5B%5D&explorer-table.advFilter=%5B%5B0,%22analytics.eventAction%22,%22BW%22,%22Page%20Translate%20~~%22,0%5D%5D

**Comment by bsmartin-ep** [comment]
Deployed to production.  Closing.
